### PR TITLE
feat: Create new product

### PR DESCRIPTION
**Pull request by trini7y**
## How should this be manually tested

#Using Postman 
- Try accessing the endpoint ```POST api/v1/products ```
- Only authenticated users have access to visit the endpoint
- If a wrong request payload is passed, it returns a response, to inform the user.

## Checklist of what I did 

- [x]  Created an endpoint ```POST /api/product``` to add a new product.
- [x]  Implemented server-side logic to handle successful and failed add product requests.
- [x]  Ensured the user has the proper permissions to add product
- [x]  Wrote unit tests for adding new products.
- [x]  Performed security testing to ensure data protection and compliance.

## Reference Issue
[[FEAT] Add Products API - BACKEND](https://github.com/hngprojects/hng_boilerplate_java_web/issues/16)

### CODE REVIEW

File: src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 import hng_java_boilerplate.product.exceptions.ValidationError;
 import hng_java_boilerplate.product.repository.ProductRepository;
+import hng_java_boilerplate.user.repository.UserRepository;
+import hng_java_boilerplate.user.entity.User;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
-import java.util.List;
 
 @Service
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java:15]: why this extra space

File: src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
 
 public class ProductServiceImpl implements ProductService{
 
-    private ProductRepository productRepository;
+    private final ProductRepository productRepository;
+    private UserRepository userRepository;
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java:19]: I can seem to find where you did dependency injection for the user repository.

File: src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java (hunk 1 above)
Comment (trini7y) [reply src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java:15]: Good morning, I will fix the extra space

File: src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java (hunk 2 above)
Comment (trini7y) [reply src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java:19]: But that's the dependency injection for the user repository

Comment (trini7y) [reply src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java]: Fixed this now

**Pull request closed without merging**